src/client/java/net/neoforged/neoforge/client/loading/NeoForgeLoadingOverlay.java

```diff
@@ -115,7 +115,11 @@ static class ExternalTexture extends AbstractTexture {
         public ExternalTexture(GpuTexture texture) {
             this.texture = texture;
             this.setFilter(false, false);
-            this.textureView = RenderSystem.getDevice().createTextureView(texture);
+            var gpuDevice = RenderSystem.getDevice();
+            if (gpuDevice instanceof ValidationGpuDevice) {
+                gpuDevice = ((ValidationGpuDevice) gpuDevice).getRealDevice();
```

Comment:
Why is this needed? I would expect `createTextureView` to be suitably implemented by `ValidationGpuDevice`.

Reply:
The loading overlay has to reach behind the validation device's back to import an external texture directly into the `GlDevice`, and as a result this then needs to do so again here because it doesn't have a `ValidationGpuTexture` instance that the `ValidationGpuDevice` is expecting but rather a `GlGpuTexture` instance.

This will be removed when the loading overlay is updated to allow it to be universal on top of B3D (see my prototype FML `APIWindow`/`APIRenderer`).

Reply:
ahh, yes, will do that